docs/how-to/how-to-build-and-test-ccv0.md

```diff
@@ -245,6 +245,82 @@ $ ~/ccv0.sh -d delete_kata_pod
 pod "nginx-kata" deleted
 ```
 
+## Verifying signed images
+
+*Note: the current proof of concept signature validation code involves hard-coding to protect a specific container* 
+*repository and is only a temporary to demonstrate the function. After the attestation agent is able to pass through*
+*trusted information and the [image management crate](https://github.com/confidential-containers/image-rs) is*
+*implemented and integrated this code will be replaced.*
+
+For the proof of concept the ability to verify images is limited to a pre-created selection of test images in our test
+repository [`quay.io/kata-containers/confidential-containers`](https://quay.io/repository/kata-containers/confidential-containers?tab=tags).
+For pulling images not in this test repo, for compatibility we fall back to the behaviour of not enforcing signatures.
+
+In our test repository there are 3 tagged images:
+- `quay.io/kata-containers/confidential-containers:signed` - this is a `busybox` image that is signed by a private GPG key
+with it's public key and signature embedded in the rootfs of the kata image.
+- `quay.io/kata-containers/confidential-containers:unsigned` - an unsigned `busybox` image
+- `quay.io/kata-containers/confidential-containers:other_signed` - an `nginx` image that is signed with a different GPG
+key and only with it's signature included in the kata rootfs, not it's public key.
+
+Along with a standard unsigned `busybox` image we can pull from `docker.io` we can test a few scenarios.
+
+From this temporary proof of concept, along with the public GPG key and signature files, a container policy file is
+created in the rootfs which specifies that any container image from `quay.io/kata-containers`
+must be signed with the embedded GPG key. In order to enable this a new agent configuration parameter called
+`policy_path` has been added which specifies the location of the policy file to use inside the image. The `ccv0.sh`
+script sets this up automatically by appending `agent.policy_file=/etc/containers/quay_verification/quay_policy.json`
+to the `kernel_params` entry in `/etc/kata-containers/configuration.toml`.
+
+With this policy parameter set a few tests of image verification can be done to test different scenarios (*Note at*
+*time of writing the `ctr shim` command has a bug, so I'm using the agent commands directly through `agent_ctl` to*
+*drive the tests*):
+- To test the fallback behaviour works using an unsigned image on an *unprotected* registry we can pull the `busybox`
+image by running:
+  ```bash
+  export CONTAINER_ID="11" && export PULL_IMAGE="docker.io/library/busybox:latest" && ~/ccv0.sh -d agent_pull_image
```

Comment:
I'd remove the `&&`'s and put this on multiple lines for clarity. Also, ooi why "`11`"? Maybe something like `my-container-id` would be clearer.

Reply:
The kuberenetes generated containerid are all hex numbers, so stuck with that. I originally have container ids of 1,2,3,4 for the 4 test cases, but when I rebased on the ali image service refactors I found that the container id length was now validated to be at least 2 characters, so bumped it up to 11, 22, 33, 44. Once I've incorporated all your feedback and re-run my testing I'll update the container ids to be more human friendly.

Reply:
> The kuberenetes generated containerid are all hex numbers

`<irony>`Wow, how user friendly!`</irony>` :smile: